From de8238e9c4ea91d323cd377e4c7c7a8dfce378fb Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 11 Aug 2023 15:23:34 +0800
Subject: [PATCH 43/71] i3c: hub: send SETHID/SETAASA ccc

Send CCCs to switch the SPD devices in slave port to i3c mode.

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 drivers/i3c/i3c-hub.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/i3c/i3c-hub.c b/drivers/i3c/i3c-hub.c
index 2955cb95da3b..82a26a5dd441 100644
--- a/drivers/i3c/i3c-hub.c
+++ b/drivers/i3c/i3c-hub.c
@@ -525,8 +525,7 @@ static int i3c_hub_configure_hw(struct device *dev)
 }
 
 static const struct i3c_device_id i3c_hub_ids[] = {
-	I3C_DEVICE(0x3c0 >> 1, 0x70, NULL),
-	//I3C_CLASS(I3C_DCR_HUB, NULL),
+	I3C_CLASS(I3C_DCR_HUB, NULL),
 	{ },
 };
 
@@ -670,6 +669,10 @@ static int i3c_hub_probe(struct i3c_device *i3cdev)
 		goto error;
 	}
 
+	if (i3cdev->bus->jesd403) {
+		i3c_device_send_ccc_cmd(i3cdev, I3C_CCC_SETHID);
+		i3c_device_send_ccc_cmd(i3cdev, I3C_CCC_SETAASA);
+	}
 	/* TBD: Apply special/security lock here using DEV_CMD register */
 
 	return 0;
-- 
2.34.1

